src/key-value-engine/introduction.md

```diff
@@ -0,0 +1,29 @@
+# Key-Value Engine
+
+A key-value engine serves at the bottommost layer in a key-value
+storage system, unless you are going to build your own file system or
+operating system. A key-value engine is crucial for a storage system
+because it manages all the persistent data directly.
+
+Most key-value engines provide some common interfaces like `Get`,
+`Put` and `Delete`. Some engines also allow you to iterate the
+key-values in order efficiently. Usually, a storage engine provides
+some special features more or less to let you take good use of it too.
+
+Choosing a key value engine is the first step to build a storage
+system. Here are some important things we need to consider:
+
+1. The data storage model. Different models are optimized for
+   different workloads. Some are good for reads and some are good for
+   writes, etc.
+2. Maturity. We don't need a storage engine to be fanny but we want it
```

Comment:
It's a great point, I do not think the distinction between "storage **system**" and "storage **engine**" is not very clear. Maybe it can be reworded?

> The performance of the storage engine limits the overall performance of the whole system. It is important to make sure the storage engine meets expectations so there is still the potential to improve the whole system.